shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -564,6 +564,10 @@ public AccessibilityNodeInfo createAccessibilityNodeInfo(int virtualViewId) {
                 granularities |= AccessibilityNodeInfo.MOVEMENT_GRANULARITY_WORD;
             }
             result.setMovementGranularities(granularities);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && semanticsNode.maxValueLength != -1) {
+                result.setMaxTextLength(semanticsNode.maxValueLength);
```

Comment:
Please double check the behavior here with non-english texts + emojis. We might need to massage the data from codePoints to graphemes or characters

Reply:
Of course this is complicated: Flutter expresses its character count as number of Unicode scalar values that are allowed (which has its own limitations, see https://master-api.flutter.dev/flutter/material/TextField/maxLength.html). Android on the other hand encodes the content of the text field with utf16 and counts the number of bytes. So the American Flag emoji has a length of 2 in Flutter, but a length of 4 for Android. A regular human being would probably consider that thing to just be one character...

Anyways, not sure how to resolve this. I'll do some more research.

Reply:
I thought about this some more. I think we really want to avoid the situation where the character counter on the flutter side is wildly out of sync with the character counter on the native side. It sounds like this would be the case with this change and there isn't much we can do about it, given that we don't control how Android/Talkback count characters.

Reply:
unless of course we could either trick Android into thinking we had a specific number of characters, or adjusted our character count to be (more?) broken

Reply:
I added another property "currentValueLength" which contains what the framework currently things the length of the entered text is. I am using this one to adjust the max value if android and flutter disagree on the size of the text. While the max value reported to android can now change depending on the characters you enter, it works great with talkback and talkback announces at the correct point that the maximum has been reached.

Reply:
🎊